help_function/src/util_group_load_store_test.cpp

```diff
@@ -0,0 +1,307 @@
+    sycl::accessor data_accessor_read_write(buffer, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(1, 1, 128), sycl::range<3>(1, 1, 128)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          auto *d_r_w =
+              data_accessor_read_write.get_multi_ptr<sycl::access::decorated::yes>()
+                  .get();
+          auto *tmp = tacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          // Load thread_data of each work item to blocked arrangement
+          group_load(tmp).load(item, d_r_w, thread_data);
+          // Store thread_data of each work item from blocked arrangement
+          group_store(tmp).store(item, d_r_w, thread_data);
+          });
+  });
+  q.wait_and_throw();
+
+  sycl::host_accessor data_accessor(buffer, sycl::read_write);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  return helper_validation_function(ptr, "test_group_load_store");
+}
+
+template <dpct::group::load_algorithm T, dpct::group::store_algorithm S> bool test_group_load_store_multiple_wgs() {
+  // Tests dpct::group::load_algorithm::BLOCK_LOAD_DIRECT 
+  // dpct::group::load_algorithm::BLOCK_LOAD_STRIPED in its entirety as API
+  // functions
+  // Tests dpct::group::store_algorithm::BLOCK_STORE_DIRECT 
+  // dpct::group::store_algorithm::BLOCK_STORE_STRIPED in its entirety as API
+  // functions
+  sycl::queue q(dpct::get_default_queue());
+  oneapi::dpl::counting_iterator<int> count_it(0);
+  sycl::buffer<int, 1> buffer(count_it, count_it + 512);
+  
+  q.submit([&](sycl::handler &h) {
+    using group_load =
+        dpct::group::workgroup_load<4, T, int, const int *, sycl::nd_item<3>>;
+    using group_store =
+        dpct::group::workgroup_store<4, S, int, int *, sycl::nd_item<3>>;
+    size_t temp_storage_size = group_load::get_local_memory_size(128);
+    sycl::local_accessor<uint8_t, 1> tacc(sycl::range<1>(temp_storage_size), h);
+    sycl::accessor data_accessor_read_write(buffer, h, sycl::read_write);
+    h.parallel_for(
+        sycl::nd_range<3>(sycl::range<3>(2, 2, 64), sycl::range<3>(1, 1, 64)),
+        [=](sycl::nd_item<3> item) {
+          int thread_data[4];
+          auto *d_r_w =
+              data_accessor_read_write.get_multi_ptr<sycl::access::decorated::yes>()
+                  .get();
+          auto *tmp = tacc.get_multi_ptr<sycl::access::decorated::yes>().get();
+          // Load thread_data of each work item to blocked arrangement
+          group_load(tmp).load(item, d_r_w, thread_data);
+          // Store thread_data of each work item from blocked arrangement
+          group_store(tmp).store(item, d_r_w, thread_data);
+          });
+  });
+  q.wait_and_throw();
+
+  sycl::host_accessor data_accessor(buffer, sycl::read_write);
+  const int *ptr = data_accessor.get_multi_ptr<sycl::access::decorated::yes>();
+  return helper_validation_function(ptr, "test_group_load_store");

```

Comment:
Can we add if it is the single work-group or multi work-group to the test string for more descriptive debug info?

Reply:
Added